### ISSUE TITLE
[package:vm_service] Missing implementations for some ext.dart.io extensions

### ISSUE THREAD

**Issue by modulovalue** [dart-lang/sdk#41456]
The [DartIOExtension](https://pub.dev/documentation/vm_service/latest/vm_service/DartIOExtension.html) extension Is missing a few key extensions such as a way to query all [open files](https://github.com/dart-lang/sdk/blob/8ef508ba3667ee817e1eee61770e529ccacec043/sdk/lib/io/file_impl.dart#L698)

```
ext.dart.io.getHttpEnableTimelineLogging, // Implemented
ext.dart.io.setHttpEnableTimelineLogging, // Implemented
ext.dart.io.getSocketProfile, // Implemented
ext.dart.io.startSocketProfiling, // Implemented
ext.dart.io.pauseSocketProfiling, // Implemented
ext.dart.io.clearSocketProfile,  // Implemented
ext.dart.io.getVersion, // Implemented
ext.dart.io.getOpenSockets, // ?
ext.dart.io.getSocketByID, // ?
ext.dart.io.getOpenFiles,  // ?
ext.dart.io.getFileByID // ?
```

Related: #41455

/cc @bkonyi

**Comment by ZichangG** [comment]
The implementation is in [sdk/lib/io/io_resource_info.dart](https://github.com/dart-lang/sdk/blob/master/sdk/lib/io/io_resource_info.dart)

**Comment by modulovalue** [comment]
@ZichangG I should've provided some context, sorry for that. I'm working on third party dart dev tooling and I'm looking for a protocol reference or for an implementation that is meant to be consumed outside of dart.io. The implementation you linked to is internal to dart.io only.

We'd like to not depend on a reverse-engineered protocol that might change any time without further notice. Since vm_service contains part of ext.dart.io, I expected to find a complete client there. This issue is about adding the missing extensions to vm_service.

**Comment by bkonyi** [comment]
These extensions will be available in `package:vm_service` `v5.1.0` once [this CL](https://dart-review.googlesource.com/c/sdk/+/157160) lands.
